frontend/src/components/user/forms.js

```diff
@@ -0,0 +1,132 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import { FormattedMessage } from 'react-intl';
+import { Form, Field } from 'react-final-form';
+
+import messages from './messages';
+import { FormSubmitButton } from '../button';
+import { pushUserDetails } from '../../store/actions/auth';
+
+const PROFILE_RELEVANT_FIELDS = [
+  'name',
+  'emailAddress',
+  'city',
+  'country',
+  'twitterId',
+  'facebookId',
+  'linkedinId',
+  'slackId',
+];
+
+function UserInterests() {
+  return (
+    <div className="cf bg-white shadow-4 pa4 mb3">
+      <h3 className="f2 mt0 fw6">
+        <FormattedMessage {...messages.welcomeTitle} />
+      </h3>
+      <p>
+        <FormattedMessage {...messages.interestsTitle} />
+      </p>
+      <p>
+        <FormattedMessage {...messages.interestsLead} />
+      </p>
+    </div>
+  );
+}
+
+class UserInformationForm extends React.Component {
+  prepareUserDetailsToPush(values) {
+    let data = {};
+    PROFILE_RELEVANT_FIELDS.filter(key => values.hasOwnProperty(key)).forEach(
+      key => (data[key] = values[key]),
+    );
+    return JSON.stringify(data);
+  }
+
+  render() {
+    const labelClasses = 'db pt3 pb2';
+    const fieldClasses = 'blue-grey w-100 mr3 pv3 ph2 input-reset ba b--grey-light bg-transparent';
+    return (
+      <div className="bg-white shadow-4 pa4">
+        <h3 className="f3 mt0 fw6">
+          <FormattedMessage {...messages.personalInfo} />
+        </h3>
+        <Form
+          onSubmit={values =>
+            this.props.pushUserDetails(this.prepareUserDetailsToPush(values), this.props.token)
+          }
+          initialValues={this.props.userDetails}
+          render={({ handleSubmit, pristine, form, submitting, values }) => {
+            return (
+              <form onSubmit={handleSubmit} className="blue-grey">
+                <div>
+                  <label className={labelClasses}>
+                    <FormattedMessage {...messages.name} />
+                  </label>
+                  <Field name="name" component="input" className={fieldClasses} />
+                </div>
+                <div>
+                  <label className={labelClasses}>
+                    <FormattedMessage {...messages.email} />
+                  </label>
+                  <Field name="emailAddress" component="input" className={fieldClasses} />
```

Comment:
`type="email"` might help enter this on mobile https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/email

This might help if it doesn't work at first:
`https://github.com/final-form/react-final-form/issues/98#issuecomment-356611506`

Reply:
Added it and also a `required` property. Thanks!